include/transaction/signer.h

```diff
@@ -10,6 +11,12 @@ class Signer
 
     std::string getSignature(std::string const &message) const;
 
+    bool verify(std::string const &signature, std::string const &message) const;
```

Comment:
> Is this in preparation for this feature?

I took a peek at that PR. However, I think that one is for signing/verifying messages with keccak ? (prefix + msgLength + msg). I think we will also need to implement that functionality in ERDCPP as well.
The reason why we implemented this is because it was required for a first version of ERDCPP to be able to:
> verify message (offchain)


> Now I realise that our signer.h shouldn't necessarily be under include/transaction/. Perhaps better under include/wallet/. However, OK for the moment.

Very good point. However, we don't have any **wallet** yet. For current version/structure, it was meant to be used only to sign transactions and messages, but as this SDK grows, we will change it

Reply:
yes for the message signing we will need the keccak hashing, so looking for a C++ library.
this PR can go as it is, the prefix concatenation + hashing can be added in another PR once we decide on the library, which then makes use of the existing signing & verification methods.